pkg/sql/schemachanger/scbuild/internal/scbuildstmt/statement_control.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package scbuildstmt
+
+import (
+	"strings"
+
+	"github.com/cockroachdb/cockroach/pkg/settings"
+	"github.com/cockroachdb/cockroach/pkg/sql/schemachanger/scerrors"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/errors"
+)
+
+// schemaStatementControl track if a statement tag is enabled or disabled
+// forcefully by the user.
+type schemaStatementControl map[string]bool
+
+// schemaChangerDisabledStatements statements which are disabled
+// for the declarative schema changer. Users can specify statement
+// tags for each statement and a "!" symbol in front can have the opposite
+// effect to force enable fully unimplemented features.
+var schemaChangerDisabledStatements = func() *settings.StringSetting {
+	return settings.RegisterValidatedStringSetting(
+		settings.TenantWritable,
+		"sql.schemachanger.disable_declarative_statement",
```

Comment:
I think combining "disable" with "!" to mean enable is confusing. Maybe rename the setting to `sql.schemachanger.force_declarative_statements` and have no prefix mean enable and `!` mean disable?